windows/src/engine/keyman32/kmprocessactions.cpp

```diff
@@ -119,58 +97,38 @@ processCapsLock(const km_core_action_item* actionItem, BOOL isUp, BOOL Updateabl
       keybd_event(VK_CAPITAL, SCAN_FLAG_KEYMAN_KEY_EVENT, KEYEVENTF_KEYUP, 0);
     }
   }
-
-  return TRUE;
 }
 
 BOOL ProcessActions(BOOL* emitKeyStroke)
 {
   PKEYMAN64THREADDATA _td = ThreadGlobals();
   if (!_td) return FALSE;
+  // TODO: #10583  Remove caching action_struct
+  if (!_td->core_actions) {
+    SendDebugMessageFormat(0, sdmGlobal, 0, "ProcessActions: core_actions not set");
+    _td->core_actions = km_core_state_get_actions(_td->lpActiveKeyboard->lpCoreKeyboardState);
+  }
 
   _td->CoreProcessEventRun = FALSE;
-  // Process the action items from the core. This actions will modify the windows context (AppContext).
-  // Therefore it is not required to copy the context from the core to the windows context.
-
-  for (auto act = km_core_state_action_items(_td->lpActiveKeyboard->lpCoreKeyboardState, nullptr); act->type != KM_CORE_IT_END; act++) {
-    BOOL continueProcessingActions = TRUE;
-    SendDebugMessageFormat(0, sdmGlobal, 0, "ProcessActions : act->type=%d", act->type);
-    switch (act->type) {
-    case KM_CORE_IT_CHAR:
-      continueProcessingActions = processUnicodeChar(_td->app, act);
-      break;
-    case KM_CORE_IT_MARKER:
-      continueProcessingActions = processMarker(_td->app, act);
-      break;
-    case KM_CORE_IT_ALERT:
-      continueProcessingActions = processAlert(_td->app);
-      break;
-    case KM_CORE_IT_BACK:
-      continueProcessingActions = processBack(_td->app, act);
-      break;
-    case KM_CORE_IT_PERSIST_OPT:
-      continueProcessingActions = processPersistOpt(act, _td->lpActiveKeyboard->lpCoreKeyboardState, _td->lpActiveKeyboard);
-      break;
-    case KM_CORE_IT_EMIT_KEYSTROKE:
-      *emitKeyStroke = TRUE;
-      continueProcessingActions = TRUE;
-      break;
-    case KM_CORE_IT_INVALIDATE_CONTEXT:
-      continueProcessingActions = processInvalidateContext(_td->app);
-      break;
-    case KM_CORE_IT_CAPSLOCK:
-      continueProcessingActions = processCapsLock(act, !_td->state.isDown, _td->TIPFUpdateable, FALSE);
-      break;
-    case KM_CORE_IT_END:
-      // fallthrough
-    default:
-      assert(false); // NOT SUPPORTED
-      break;
-    }
-    if (!continueProcessingActions) {
-      return FALSE;
-    }
+
+
+
+  processBack(_td->app, _td->core_actions->code_points_to_delete, _td->core_actions->deleted_context);
+  processUnicodeChar(_td->app, _td->core_actions->output);
+  if (_td->core_actions->persist_options != NULL) {
+    processPersistOpt(_td->core_actions, _td->lpActiveKeyboard);
+  }
+  if (_td->core_actions->do_alert) {
+    processAlert(_td->app);
   }
+  if (_td->core_actions->emit_keystroke) {
+    *emitKeyStroke = TRUE;
```

Comment:
```suggestion
    *emitKeystroke = TRUE;
```

(Keystroke is one word)